lib/reconcile/astarte_dashboard.go

```diff
@@ -163,7 +163,20 @@ func getAstarteDashboardConfigMapData(cr *apiv1alpha1.Astarte, dashboard apiv1al
 	v := getSemanticVersionForAstarteComponent(cr, dashboard.Version)
 	constraint, _ := semver.NewConstraint("< 0.11.0")
 
-	if dashboard.Config.RealmManagementAPIURL == "" {
+	constraint10, _ := semver.NewConstraint(">= 1.0.0")
+	isAstarte10 := constraint10.Check(v)
+
+	if isAstarte10 {
+		// Astarte 1.0+ just needs astarte_api_url, and single API urls only if they're explicit
+		dashboardConfig["astarte_api_url"] = getBaseAstarteAPIURL(cr)
+		// We also enable Flow preview for Astarte 1.0+
+		dashboardConfig["enable_flow_preview"] = true
+	} else {
+		// secure_connection is needed only for Astarte pre-1.0
+		dashboardConfig["secure_connection"] = pointy.BoolValue(cr.Spec.API.SSL, true)
+	}
+
+	if dashboard.Config.RealmManagementAPIURL == "" && !isAstarte10 {
```

Comment:
"realm_management_api_url" and other URLs are still needed if the path is different from the canonical one (base URL + "/realmmanagement" etc etc)

Reply:
That's the `else` branch: if the API url is explicitly given (_i.e._ `dashboard.Config.RealmManagementAPIURL != ""`), then `dashboardConfig["realm_management_api_url"]` is always explicitly set (there's no version constraints in the `else` branch).